### PR TITLE
replace dayjs with luxon

### PR DESCRIPTION
**Pull request by ydomenjoud**
wait upgrade with angular 11

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts [hunk 1]
```diff
@@ -125,7 +128,7 @@ export class KalDatepickerComponent extends FormElementComponent<KalDate> implem
     } else if (this.isCurrentDateValid) {
       return this.currentDate.getYear() + this.yearsIncrement;
     } else {
-      return dayjs().year() + this.yearsIncrement;
+      return KalDate.now().year + this.yearsIncrement;
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts]: Tu peux utiliser new KalDate().year

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts [hunk 2]
```diff
@@ -342,4 +310,38 @@ export class KalDatepickerComponent extends FormElementComponent<KalDate> implem
...
+          date = new KalDate();
+        }
+
+        this.currentDate = date;
+      })
+    ).subscribe();
+    this.subscriptions.push(valueChangesSubscription);
+
+    const focusOnKalInputSubscription = fromEvent<MouseEvent>(this.kalInput.inputElement.nativeElement, 'focus')
+      .subscribe(() => this.open());
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts]: Utilise un tap pour cet observable

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

File: projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts [hunk 3]
```diff
@@ -65,6 +64,8 @@ export class KalDatepickerComponent extends FormElementComponent<KalDate> implem
    */
   currentDate: KalDate;
 
+  @Input() format: KalDateFormat;
```
Comment (coco33300) [inline projects/kalidea/kaligraphi/src/lib/02-form/kal-datepicker/kal-datepicker.component.ts:67]: This input doesn't seem to be used in the component

File: projects/kalidea/kaligraphi/src/lib/99-utility/kal-date/kal-date.ts [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+  private static getOption<U extends keyof KalDateOptions, R>(key: U, fallback?: KalDateOptions[U]): KalDateOptions[U] {
+    return KalDateModule.kalDateOptions?.[key] || fallback;
+  }
+
+  /**
+   * Whether the KalDate is valid.
+   */
+  get valid(): boolean {
+    return this.value && this.value.isValid;
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/99-utility/kal-date/kal-date.ts]: return this.value?.isValid;

File: projects/kalidea/kaligraphi/src/lib/99-utility/kal-date/kal-date.ts [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+  isSameOrAfter(date: KalDateType, unit: KalDurationUnit = 'days'): boolean {
+    const comparisonDate = KalDate.getDate(date);
+    return this.isSame(comparisonDate, unit) || this.isAfter(comparisonDate, unit);
+  }
+
+  /**
+   * return true if current date is today
+   */
+  isToday(): boolean {
+    return this.isSame(KalDate.now());
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/99-utility/kal-date/kal-date.ts]: Tu peux utiliser un new KalDate()

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

**Review by MichaelGrosio — APPROVED** [review]

**Pull request merged**